src/java/org/apache/cassandra/io/util/RandomAccessReader.java

```diff
@@ -71,6 +73,68 @@ private void reBufferAt(long position)
         assert buffer.order() == ByteOrder.BIG_ENDIAN : "Buffer must have BIG ENDIAN byte ordering";
     }
 
+    public void readFully(float[] dest) throws IOException
```

Comment:
we have a wrapper class for RAR elsewhere in this patch, but here we're implementing most of it directly, do we need the wrapper?

Reply:
I realised that when I updated JVector to 1.0.2 and added more code in this class. The original reason I added here was because `bufferHolder` and the constructor were private.

I've changed this and pulled the `RandomAccessReaderAdapter` out into the `vector` package and made it extend `RandomAccessReader`. This looks a lot cleaner and limits the changes here to making `bufferHolder` and the constructor protected.